### PR TITLE
Allow voting on proposals up to completion

### PR DESCRIPTION
**Pull request by 0xekez**
Also updates the frontend code IDs to use the [latest release](https://github.com/DA0-DA0/dao-contracts/releases/tag/v2.0.0-beta).

ref: https://github.com/DA0-DA0/dao-contracts/commit/a6e01eacbe3a50ff867d68cf0d2c27aeb4f7f649

### CODE REVIEW

File: packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/components/ProposalStatusAndInfo.tsx [hunk 1]
```diff
@@ -212,6 +212,9 @@ const InnerProposalStatusAndInfo = ({
       : []),
   ]
 
+  const proposalCompleted =
+    proposal.status !== Status.Open && proposal.status !== Status.Passed
```
Comment (NoahSaso) [inline packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/components/ProposalStatusAndInfo.tsx:216]: this means that a proposal is only completed (and thus disallows votes cast) when it is rejected, executed(/failed execution), or closed. looking at the contract code, it seems like even passed proposals can be voted on before expiry. am I misunderstanding that, or is there a reason you are preventing passed proposals from being voted on here?

i would think the condition here should be `NOT open AND expired`.

also nit: `proposalCompleted` here is a bit vague as there's additional context relevant here; an explanatory comment might help, or renaming to something like `proposalOpenForVoting`

File: packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/hooks/useLoadingWalletVoteInfo.tsx [hunk 2]
```diff
@@ -80,18 +80,18 @@ export const useLoadingWalletVoteInfo = ():
     totalVotingPowerWhenProposalCreatedLoadable.contents.power
   )
 
+  const proposalCompleted =
+    proposal.status !== Status.Open && proposal.status !== Status.Passed
```
Comment (NoahSaso) [inline packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/hooks/useLoadingWalletVoteInfo.tsx:84]: same as other comment

File: packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/hooks/useLoadingWalletVoteInfo.tsx [hunk 3]
```diff
@@ -80,18 +80,18 @@ export const useLoadingWalletVoteInfo = ():
...
+  const canVote =
+    couldVote && !proposalCompleted && (!walletVote || proposal.allow_revoting)
```
Comment (NoahSaso) [inline packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/hooks/useLoadingWalletVoteInfo.tsx:86]: love how declarative this is ❤️

File: packages/utils/constants/codeIdConfigs.ts [hunk 4]
```diff
@@ -2,21 +2,21 @@ import { CodeIdConfig } from '@dao-dao/types'
...
+  DaoCore: 3457,
+  DaoPreProposeMultiple: 3460,
+  DaoPreProposeSingle: 3461,
+  DaoProposalMultiple: 3462,
+  DaoProposalSingle: 3463,
+  DaoVotingCw20Staked: 3464,
+  DaoVotingCw4: 3465,
+  DaoVotingCw721Staked: 3466,
+  DaoVotingNativeStaked: 3467,
+  CwTokenSwap: 3456,
```
Comment (NoahSaso) [inline packages/utils/constants/codeIdConfigs.ts:19]: 🥵😫😤🫡 hell yeah

**Review by NoahSaso — CHANGES_REQUESTED** [review]
awesome 🔥 just one thing

File: packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/components/ProposalStatusAndInfo.tsx (hunk 1 above)
Comment (0xekez) [reply packages/stateful/proposal-module-adapter/adapters/DaoProposalSingle/components/ProposalStatusAndInfo.tsx:216]: ah you're right. thanks!

**Comment by NoahSaso** [comment]
also I just realized this feature can only activate if the `coreVersion` of the DAO is not V1. (this can be retrieved from the `useDaoInfoContext` hook)

**Review by NoahSaso — APPROVED** [review]

**Pull request merged**